### ISSUE TITLE
Using authenticated calls

### ISSUE THREAD

**Issue by ferigis** [zhongwencool/eetcd#4]
Hi!

We want to use authenticated calls with `eetcd`. We are able to get the `token` calling `:eetcd_auth.authenticate/1` but how we can use that token for further calls to the KV? is there any example?

Thanks!

**Comment by zhongwencool** [comment]
@ferigis I have no experience with this. But I find [this document from](https://github.com/etcd-io/etcd/blob/master/Documentation/dev-guide/api_grpc_gateway.md)
```
# get the auth token for the root user
curl -L http://localhost:2379/v3/auth/authenticate \
  -X POST -d '{"name": "root", "password": "pass"}'
# {"header":{"cluster_id":"14841639068965178418","member_id":"10276657743932975437","revision":"1","raft_term":"2"},"token":"sssvIpwfnLAcWAQH.9"}
```
```
curl -L http://localhost:2379/v3/kv/put \
  -H 'Authorization : sssvIpwfnLAcWAQH.9' \
  -X POST -d '{"key": "Zm9v", "value": "YmFy"}'
# {"header":{"cluster_id":"14841639068965178418","member_id":"10276657743932975437","revision":"2","raft_term":"2"}}
```

------
So I have not implemented this method yet, I just hard code this header  :(
https://github.com/zhongwencool/eetcd/blob/master/src/eetcd_stream.erl#L9

I will change this a few days later.

**Comment by zhongwencool** [comment]
@ferigis  Plz try `0.2.1`. 
```erlang
AuthRequest = #'Etcd.AuthenticateRequest'{name = <<"youruser">>, password = <<"yourpassword">>},
{ok, #'Etcd.AuthenticateResponse'{token = Token}} = eetcd_auth:authenticate(AuthRequest),
{ok, PutResponse} = eetcd_kv:put(#'Etcd.PutRequest'{key = <<"key">>, value = <<"value">>}, Token]).
%% or 
{ok, PutResponse} = eetcd_kv:put(#'Etcd.PutRequest'{key = <<"key">>, value = <<"value">>}, [{<<"Authorization">>, Token}]).
```

**Comment by seudin** [comment]
The correct API call is:

```{ok, PutResponse} = eetcd_kv:put(#'Etcd.PutRequest'{key = <<"key">>, value = <<"value">>}, [{<<"authorization">>, Token}]).```

Note, header field names MUST be converted to the lowercase see:
https://tools.ietf.org/html/rfc7540#section-8.1.2

Since gun library and underlining cowlib doesn't convert user's header to the lowercase, the API call breaks HTTP2 protocol.

To use API without override header, all ```<<"Authorization">>``` in eetcd must be converted to lowercase.

**Comment by zhongwencool** [comment]
Thanks for reminding this. I will correct it tomorrow and update the readme.

**Comment by zhongwencool** [comment]
fixed on v0.2.2
